packages/tocco-ui/src/EditableValue/typeEditors/Duration.js

```diff
@@ -0,0 +1,124 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+
+class Duration extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      ...this.millisecondsToDuration(props.value)
+    }
+  }
+
+  millisecondsToDuration = milliSeconds => {
+    if (!milliSeconds && milliSeconds !== 0) {
+      return {hours: '', minutes: ''}
+    }
+
+    const minutes = parseInt((milliSeconds / (1000 * 60)) % 60)
+    const hours = parseInt((milliSeconds / (1000 * 60 * 60)) % 24)
+    return {hours, minutes}
+  }
+
+  handleHourChange = e => {
+    const hours = e.target.value.replace(/[^\d]/, '')
+    this.setState({...this.state, hours})
+    this.handleChange(hours, null)
+  }
+
+  handleMinutesChange = e => {
+    let minutes = e.target.value.replace(/[^\d]/, '')
+    
+    if (!e.target.validity.valid) {
+      minutes = 0
+    }
+
+    if (`${minutes}`.length > 2) {
+      minutes = `${minutes}`.slice(0, 2)
+    }
+
+    if (minutes > 59) {
+      minutes = 0
+    }
+
+    this.setState({...this.state, minutes: minutes})
+
+    this.handleChange(null, minutes)
+
+    if (`${minutes}`.length === 2 || minutes === 0) {
+      e.target.select()
+    }
+  }
+
+  handleChange = (hours, minutes) => {
+    const minutesValue = minutes !== null ? minutes : this.state.minutes
+    const hoursValue = (hours !== null ? hours : this.state.hours)
+    if (minutesValue === '' && hoursValue === '') {
+      this.props.onChange(null)
+      return
+    }
+
+    const hoursMilliseconds = (hoursValue || 0) * 60 * 60000
```

Comment:
maybe move calculation and addition in seperate function and call onChange with the result in handleChange